src/economy/economy.hpp

```diff
@@ -155,15 +167,22 @@ float pop_farmer_min_wage(sys::state& state, dcon::nation_id n, float min_wage_f
 float pop_laborer_min_wage(sys::state& state, dcon::nation_id n, float min_wage_factor);
 float pop_factory_min_wage(sys::state& state, dcon::nation_id n, float min_wage_factor);
 
-std::tuple<float, float, float> rgo_relevant_population(sys::state& state, dcon::province_id p, dcon::nation_id n);
+struct rgo_workers_breakdown {
+	float paid_workers;
+	float slaves;
+	float total;
+};
+
+rgo_workers_breakdown rgo_relevant_population(sys::state& state, dcon::province_id p, dcon::nation_id n);
+
 float rgo_overhire_modifier(sys::state& state, dcon::province_id p, dcon::nation_id n);
 float rgo_desired_worker_norm_profit(sys::state& state, dcon::province_id p, dcon::nation_id n, float min_wage, float total_relevant_population);
 float rgo_expected_worker_norm_profit(sys::state& state, dcon::province_id p, dcon::nation_id n);
 
 
 void update_rgo_employment(sys::state& state);
 void update_factory_employment(sys::state& state);
-void daily_update(sys::state& state);
+void daily_update(sys::state& state, bool initiate_building);
```

Comment:
you could have done this instead: `void daily_update(sys::state& state, bool initiate_building = true);` this would give the new parameter a default value, meaning that in old code, where it was not present, it would be treated as being passed `true`. Of course, this isn't helpful now, since you have already added the new parameter eveywhere